src/whir/committer/reader.rs

```diff
@@ -242,37 +270,70 @@ mod tests {
         // Create WHIR config with 5 variables and 3 OOD samples, plus a random number generator.
         let (params, mut rng) = make_test_params(5, 3);
 
+        // Get the underlying whir_params for proof initialization
+        let whir_params = ProtocolParameters {
+            initial_statement: true,
+            security_level: 100,
+            pow_bits: 10,
+            rs_domain_initial_reduction_factor: 1,
+            folding_factor: FoldingFactor::ConstantFromSecondRound(4, 4),
+            merkle_hash: params.merkle_hash.clone(),
+            merkle_compress: params.merkle_compress.clone(),
+            soundness_type: SecurityAssumption::CapacityBound,
+            starting_log_inv_rate: 1,
+            univariate_skip: false,
+        };
+
         // Create a random degree-5 multilinear polynomial (32 coefficients).
         let polynomial = EvaluationsList::new((0..32).map(|_| rng.random()).collect());
 
+        // Initialize the Whir Proof structure
+        let mut proof = WhirProof::from_protocol_parameters(&whir_params, 5);
+
         // Instantiate the committer using the test config.
         let committer = CommitmentWriter::new(&params);
 
         // Use a DFT engine to expand/fold the polynomial for evaluation.
         let dft = EvalsDft::default();
 
-        // Set up Fiat-Shamir transcript and commit the protocol parameters.
-        let mut ds = DomainSeparator::new(vec![]);
-        ds.commit_statement::<_, _, _, 8>(&params);
-
-        // Create the prover state from the transcript.
+        // Set up the challenger and observe protocol parameters
         let mut rng = SmallRng::seed_from_u64(1);
-        let challenger = MyChallenger::new(Perm::new_from_rng_128(&mut rng));
+        let mut challenger = MyChallenger::new(Perm::new_from_rng_128(&mut rng));
+
+        // Observe protocol parameters for domain separation
+        challenger.observe(F::from_u64(proof.rounds.len() as u64));
+        challenger.observe(F::from_u64(proof.final_queries.len() as u64));
```

Comment:
I think that the same suggestion applies to many places of the codebase where you can use from_usize directly especially in tests to simplify

```suggestion
        challenger.observe(F::from_usize(proof.rounds.len()));
        challenger.observe(F::from_usize(proof.final_queries.len()));
```